full-stack-tests/backend/src/integration/CloudSqlite.test.ts

```diff
@@ -188,6 +188,47 @@ describe("CloudSqlite", () => {
     container.disconnect({detach: true});
   });
 
+  it("should LogLevel.Trace set LogMask to ALL", async () => {
+    Logger.initializeToConsole();
```

Comment:
Rather than spying on console.log, You could pass functions like sinon.spy() to Logger.initialize

`const logTrace = sinon.spy()`
`const loginfo = sinon.spy()`
`Logger.initialize(undefined,. undefined, logInfo, logTrace)`

And then I think you could assert those were called with what you expect.

At the end of the test you should probably also call Logger.initialize() (in addition to any cleaning of the spies you might have to do) to clear up any spies setup in the logger.

Reply:
resolved in [commit d3af9c8c5a565d029d9d040de8e17c3384727671](https://github.com/iTwin/itwinjs-core/pull/6973/commits/d3af9c8c5a565d029d9d040de8e17c3384727671)

Reply:
I meant you could just do sinon.spy() with nothing inside, did you find that didn't work? either way what you switched to is fine. Resolving

Reply:
sinon.spy() didn't work. We need a `LogFunction` in `Logger.initialize()`

Reply:
Hmm maybe could cast it `as LogFunction`? good to know though.

Reply:
Yes. But we also need to assert that certain messages get logged, for instance, "dirty block".  With just `sinon.spy()`, we cannot do that.

Reply:
I assumed that you could still do `spy.getCalls().some((call) =>call.args[1].includes("is now dirty block"));` but I could be wrong